src/inference_profiler.cc

```diff
@@ -623,12 +623,15 @@ InferenceProfiler::Profile(
   meets_threshold = true;
 
 
-  RETURN_IF_ERROR(dynamic_cast<RequestRateManager*>(manager_.get())
-                      ->PerformWarmup(request_rate, warmup_request_count));
+  if (dynamic_cast<CustomRequestScheduleManager*>(manager_.get()) == nullptr) {
+    std::cout << "Perform Warmup" << std::endl;
+    RETURN_IF_ERROR(dynamic_cast<RequestRateManager*>(manager_.get())
+                        ->PerformWarmup(request_rate, warmup_request_count));
+    std::cout << "Request Rate: " << request_rate
+              << " inference requests per second" << std::endl;
+  }
```

Comment:
Can this be rewritten to be more readable? I can't easily understand what's going on. Too many negations and special conditions and logic that happens regardless.